lib/ui/painting.dart

```diff
@@ -1607,12 +1655,152 @@ class Path extends NativeFieldWrapperClass2 {
   /// Returns a copy of the path with all the segments of every
   /// subpath transformed by the given matrix.
   Path transform(Float64List matrix4) {
-    assert(matrix4 != null);
-    if (matrix4.length != 16)
-      throw new ArgumentError('"matrix4" must have 16 entries.');
+    assert(_matrix4IsValid(matrix4));
     return _transform(matrix4);
   }
   Path _transform(Float64List matrix4) native 'Path_transform';
+
+  /// Gets the bounding rectangle for this path.
+  Rect getBounds() {
+    final Float32List rect = _getBounds();
+    return new Rect.fromLTRB(rect[0], rect[1], rect[2], rect[3]);
+  }
+  Float32List _getBounds() native 'Path_getBounds';
+
+  /// Sets this path to the result of operation [PathOperation]
+  /// 
+  /// If only one path is specified, this path will be used as the first 
+  /// operand, otherwise path1 will be the first operand and path2 the
+  /// second.
+  /// 
+  /// The resulting path will be constructed from non-overlapping contours. The
+  /// curve order is reduced where possible so that cubics may be turned into
+  /// quadratics, and quadratics maybe turned into lines.
+  static Path combine(PathOperation operation, Path path1, Path path2) {
+    assert(path1 != null);
+    assert(path2 != null);
+
+    final Path path = new Path();
+    if (path._op(path1, path2, operation.index)) {
+      return path;
+    } 
+    throw new StateError('Path combine failed - likely due to invalid path');
+  }
+  bool _op(Path path1, Path path2, int operation) native 'Path_op';
+
+  /// Creates a [PathMetrics] object for this path.
+  PathMetrics computeMetrics({bool forceClosed = false}) {
```

Comment:
In a second paragraph, explain what the argument does.

Also, ` = ` -> `: ` for consistency with flutter style

Reply:
There's a note in the langauge tour suggesting that the `:` would be deprecated (assuming you're looking for `{bool forceClosed: false}`:

>Version note: Old code might use a colon (:) instead of = to set default values of named parameters. The reason is that before SDK 1.21, only : was supported for named parameters. That support is likely to be deprecated, so we recommend that you use = to specify default values, and specify an SDK version of 1.21 or higher.
https://www.dartlang.org/guides/language/language-tour#optional-parameters

Should I still change it?

Reply:
Yeah, we're using `:` everywhere for now, until we get around to switching everything over at once. Keeps things consistent in the meantime.